src/attach.ts

```diff
@@ -108,7 +108,7 @@ function initNodeTable(nodeTable: NodeTable, node: YamlUnistNode): void {
   }
 
   if ("children" in node) {
-    (node.children as Array<(typeof node.children)[number]>).forEach(child => {
+    (node.children as Array<typeof node.children[number]>).forEach(child => {
```

Comment:
@thorn0 @sosukesuzuki Is this expected? As one not familiar with TS, it's unreadable to me. I can't find when we changed this.

Reply:
Noticed this too. Looks bad.

Reply:
Pretty sure it was me in https://github.com/prettier/prettier/pull/6605, not intentionally.

Reply:
Strange thing is nobody complain about it for 3 years. I even thought this changes syntax.